src/node_dir.h

```diff
@@ -0,0 +1,87 @@
+#ifndef SRC_NODE_DIR_H_
+#define SRC_NODE_DIR_H_
+
+#if defined(NODE_WANT_INTERNALS) && NODE_WANT_INTERNALS
+
+#include "node_file.h"
+#include "node.h"
+#include "req_wrap-inl.h"
+
+namespace node {
+
+using v8::Context;
+using v8::FunctionCallbackInfo;
+using v8::HandleScope;
+using v8::Local;
+using v8::MaybeLocal;
+using v8::Null;
+using v8::Object;
+using v8::Persistent;
+using v8::Promise;
+using v8::Value;
+
+using fs::FSReqBase;
+using fs::FSReqAfterScope;
+using fs::FSReqWrapSync;
+
+namespace fs_dir {
+
+class DirHandle;
+class DirReq;
+
+// XXX Needed to have access to the DirHandle in async opendir so that we can
+// set the uv_dir_t.
```

Comment:
Can you explain this comment a bit more? I don’t really see an issue with this approach, but the `XXX` seems to hint at one?